### ISSUE TITLE
Extracting longer works with the v3.2.3.1595 release

### ISSUE THREAD

**Issue by philipsd6** [ricklupton/rmscene#5]
I'm not sure it's helpful to get bug reports from the bleeding edge release, but just in case:

```python
poetry run python -m rmscene print-blocks \
    ~/backup/remarkable/latest/files/7de2ff35-356b-425e-bba1-5b1e0b5a7f94/809e181d-8974-461c-a824-da0ec4fb0713.rm

AuthorIdsBlock(author_uuids={1: UUID('624b4da6-5190-505d-ac2a-947f7e7505cf')})

MigrationInfoBlock(migration_id=CrdtId(1, 1), is_device=True)
Block starting at 84, length 7, only read 5
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/philipsd6/devel/remarkable/rmscene/src/rmscene/__main__.py", line 62, in <module>
    args.func(args)
  File "/home/philipsd6/devel/remarkable/rmscene/src/rmscene/__main__.py", line 35, in pprint_file
    for el in result:
  File "/home/philipsd6/devel/remarkable/rmscene/src/rmscene/scene_stream.py", line 738, in read_blocks
    yield from _read_blocks(stream)
  File "/home/philipsd6/devel/remarkable/rmscene/src/rmscene/scene_stream.py", line 723, in _read_blocks
    yield block_type.from_stream(stream)
  File "/home/philipsd6/devel/remarkable/rmscene/src/rmscene/scene_stream.py", line 99, in from_stream
    migration_id = stream.read_id(1)
  File "/home/philipsd6/devel/remarkable/rmscene/src/rmscene/tagged_block_reader.py", line 56, in read_id
    self.data.read_tag(index, TagType.ID)
  File "/home/philipsd6/devel/remarkable/rmscene/src/rmscene/tagged_block_common.py", line 93, in read_tag
    raise UnexpectedBlockError(
rmscene.tagged_block_common.UnexpectedBlockError: Expected index 1, got 0, at position 97
```

**Comment by valentin-ricard** [comment]
Can replicate in 3.2.2.1581 (latest stable on rM2)

**Comment by valentin-ricard** [comment]
Also, here is a file that causes an error (managed to get the unknown value to something else than 0, by opening the file in the desktop app and just double-clicking to begin to add text, but not actually writing.
(zipped the file due to github not supporting .rm files)
[2d23993e-0422-4360-bce5-6c3e97a47d21.zip](https://github.com/ricklupton/rmscene/files/10888500/2d23993e-0422-4360-bce5-6c3e97a47d21.zip)

**Comment by ricklupton** [comment]
@philipsd6 thanks for the report, the error should be fixed now I think as in #6
